Sources/SMTPKitten/SMTPClient.swift

```diff
@@ -316,15 +324,51 @@ public final class SMTPClient {
         try await self.channel.pipeline.addHandler(sslHandler, position: .first)
     }
     
-    public func login(user: String, password: String) async throws {
-        try await send(.authenticateLogin)
-            .status(.containingChallenge, or: SMTPError.loginFailure)
-        
-        try await send(.authenticateUser(user))
-            .status(.containingChallenge, or: SMTPError.loginFailure)
-        
-        try await self.send(.authenticatePassword(password))
-            .status(.authSucceeded, or: SMTPError.loginFailure)
+    public func login(user: String, password: String, force method: SMTPAuthMethod? = nil) async throws {
```

Comment:
```suggestion
    /// Authenticates to the SMTP server. If `method` is `nil`, the method will be chosen automatically. Otherwise
    /// the requested method will be used
    public func login(user: String, password: String, method: SMTPAuthMethod? = nil) async throws {
```